source/retryable-writes/retryable-writes.rst

```diff
@@ -52,65 +51,57 @@ Transaction ID
+   Session specification. This object is always associated with a server
+   session; however, drivers will pool server sessions so that creating a
+   DriverSession will not always entail creation of a new server session. The
+   name of this object MAY vary across drivers.
 
 Additional terms may be defined in the Driver Session specification.
 
 Naming Deviations
 -----------------
 
-This specification defines the name for a new DriverSession option,
-``retryWrites``. To the extent possible, drivers SHOULD use the defined name.
-However, where a driver’s or language’s naming conventions would conflict,
-drivers SHOULD honor their existing conventions. For example, a driver may use
-``retry_writes`` instead of ``retryWrites``.
+This specification defines the name for a new MongoClient option,
+``retryWrites``. Drivers MUST use the defined name to ensure portability of
+connection strings across applications and drivers.
 
-Configuration
--------------
+MongoClient Configuration
+-------------------------
 
-The following option MUST be added to the SessionOptions data structure:
+This specification introduces the following client-level configuration option.
 
-.. code:: typescript
+retryWrites
+~~~~~~~~~~~
 
-  class SessionOptions {
-    /**
-     * If true, retryable behavior will be applied to write commands
-     * executed within the session.
-     *
-     * Defaults to false, which implies no change in write behavior.
-     */
-    retryWrites: Optional<Boolean>;
-  }
+This boolean option determines whether retryable behavior will be applied to all
+write operations executed within the application. This option MUST default to
+false, which implies no change in write behavior.
+
+This option MUST NOT be configurable at the level of a database object,
+collection object, or at the level of an individual write operation.
 
 Generating Transaction IDs
 --------------------------
 
 The server requires each retryable write operation to provide a unique
 transaction ID in its command document. The transaction ID consists of a server
 session ID and a monotonically increasing transaction number. The session ID is
-obtained when creating the server session. Drivers will be responsible for
-maintaining a monotonically increasing transaction number for each server
-session used by a DriverSession object created with the ``retryWrites`` option.
-Drivers that pool server sessions MUST preserve the transaction number when
-reusing a server session from the pool with a new DriverSession.
+obtained from the DriverSession object, which will have either been passed to
+the write operation from the application or constructed internally for the
+operation. Drivers will be responsible for maintaining a monotonically
+increasing transaction number for each server session used by a DriverSession
+object. Drivers MUST preserve the transaction number when reusing a server
+session from the pool with a new DriverSession.
```

Comment:
https://github.com/mongodb/specifications/pull/174/commits/61eb2db2283f5bd91f3af03acbe10334cf18ebfb adds a suggestion to track transaction number on the server session object.